docs/polkadot/networking/3-attestation.md

```diff
@@ -1,54 +1,268 @@
+
+A validator ring is mostly-connected as permitted by the physical topology. Nodes within this ring talk to each other periodically via short-term and low-cost QUIC connections.
+
+For a given parachain, the primary checkers are also mostly-connected, as permitted by the physical topology. Likewise the secondary checkers are also mostly-connected.
+
+These connections represent the vast majority of traffic flow in our A&V networking protocol; to improve reliability and availability there are also other lines of communication as described below.
+
+The protocol runs in several phases and stages. Every node acts both in the distributor and distributee role, but not every role is active in every stage. A summary follows:
+
+Phase | Distributors | Distributees
+----- | ------------ | ------------
+P1SA  | Y            | N
+P1SB  | Y            | Y
+P2SA  | Y            | N
+P2SB  | N            | Y
+
+Note that there is also background activity, as described below.
+
+## Protocol phase 1: initial distribution
+
+Every validator is both a distributor of C pieces and a distributee (recipient) of C pieces [TODO: correction, explanation]. Every piece has one source parachain and one main target storer, and so we can index pieces with a tuple (c_s, (c_t, i_t)).
+
+In phase 1, pieces are distributed by the source parachain validators (c_s, \*) to the main target storers. This happens in two stages. Stage A is where most of the material is distributed, and stage B acts as a backup mechanism for anything that was missed during stage A.
+
+**Stage A**
+
+As a distributor, each validator (c, i) attempts to send the relevant pieces for their parachain's block (c, (c', i)), to everyone else on their ring (c', i) for all c' != c. Conversely as a distributee, each validator expects to receive their relevant pieces (c', (c, i)) for other parachains' blocks, from everyone else on their ring (c', i) for all c' != c.
+
+In more detail:
+
+Each distributor (c, i) will, with parallelism = C / 4, for s in [0..C), try to send the relevant piece to target t = (c+s, i) [TBX S1 #2]. C / 4 comes from our estimate that T_b ~= 4 * T_L.
+
+Trials are done with a timeout, slightly larger than T_l. Sending is via QUIC. In order for it to be treated as a success, it should include an acknowledgement of receipt. Note this is orthogonal from the gossiped receipts which include a validator signature; by contrast this transport-level receipt can be assumed to be already protected by QUIC transport authentication.
```

Comment:
> by QUIC transport authentication

As discussed in the past today there is no notion matching chain identity with network identity. I remember you had some design document for authentication. Maybe worth linking here.

Reply:
Fixed in the latest commit.